src/helpers.ts

```diff
@@ -44,16 +44,18 @@ export async function retry<T>(
   job: () => Promise<T>,
   { retries = 5, retryDelayMS = 0 } = {},
 ): Promise<T> {
-  let err: Error | null = null;
+  let err: any = null;
   for (let i = 0; i < retries; i++) {
     try {
       const res = await job();
       return res;
-    } catch (e) {
-      err = e;
-      // wait before retrying if it's not the last try
-      if (retryDelayMS && i < retries - 1) {
-        await new Promise((r) => setTimeout(r, retryDelayMS));
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        err = e;
+        // wait before retrying if it's not the last try
+        if (retryDelayMS && i < retries - 1) {
+          await new Promise((r) => setTimeout(r, retryDelayMS));
+        }
```

Comment:
seems that it will silently success if it throw something else than an error
is it expected?

Reply:
Updated accordingly.